### PR TITLE
Small performance boosts

### PR DESCRIPTION
**Pull request by DanielNoord**
## Steps

- [x] For new features or bug fixes, add a ChangeLog entry describing what your PR does.
- [x] Write a good description on what the PR does.

## Description

Kept playing around with some profiles. This is a very minor **-0.5%** boost, but that's probably still worth it.

## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :hammer: Refactoring   |

### CODE REVIEW

File: astroid/nodes/node_ng.py [hunk 1]
```diff
@@ -174,17 +174,15 @@ def infer(
...
         # exponentially exploding possible results.
-        limit = AstroidManager().max_inferable_values
-        for i, result in enumerate(generator):
+        limit = AstroidManager.max_inferable_values
+        for i, result in enumerate(self._infer(context=context, **kwargs)):
             if i >= limit or (context.nodes_inferred > context.max_inferred):
-                uninferable = util.Uninferable
-                results.append(uninferable)
-                yield uninferable
+                results.append(util.Uninferable)
```
Comment (Pierre-Sassoulas) [inline astroid/nodes/node_ng.py:184]: I don't understand why we yield and also add the value to a results list ? Shouldn't it be one or the other ?

Comment (DanielNoord) [reply astroid/nodes/node_ng.py:184]: Results is being used later to store the results to the cache.

**Review by Pierre-Sassoulas — APPROVED** [review]

File: astroid/transforms.py [hunk 2]
```diff
@@ -32,9 +32,6 @@ def _transform(self, node: NodeNG) -> NodeNG:
         transformed node.
         """
         cls = node.__class__
-        if cls not in self.transforms:
-            # no transform registered for this class of node
-            return node
 
         transforms = self.transforms[cls]
```
Comment (Pierre-Sassoulas) [inline astroid/transforms.py:36]: Is it safe to do that after removing the condition ?

Comment (DanielNoord) [reply astroid/transforms.py:36]: Yes, `transforms` is a `defaultdict` so the `if` statement is completely unnecessary.

_Performance benefit is minimal but I found it when looking at this method. Might give us a 0.00001% boost 😄_

**Review by Pierre-Sassoulas — APPROVED** [review]

**Pull request merged**